cmd/run.go

```diff
@@ -124,7 +135,11 @@ var RunCmd = &cobra.Command{
 					os.Exit(1)
 				}
 
-				daprRunning <- true
+				//if app exposes service, don't wait for daprd
+				if appPort != -1 {
```

Comment:
I think a better approach to this would be in Dapr runtime to start http, grpc servers early in its lifecycle.

Reply:
i totally agree this is just hack. instead of this, starting http/grpc listeners earlier will be better, increase chances for conn success ( although 100% success can not quaranteed? since daprd & app(without service) start ups are not synced & nondeterministic.  right now, go-sdk, java-sdk just contain protobuf stub code; is this intended minimal design? if in future it will provide wrappers for rest api, then could add conn-retry code at app side in sdk init()/conn()?

Reply:
looks like dapr sidecar startup in pod can be solved by KEP (https://github.com/kubernetes/enhancements/blob/master/keps/sig-apps/sidecarcontainers.md). For local startup issue, i saw this dapr issue 725 (https://github.com/dapr/dapr/issues/725) for adding service health check endpoint also required for above-mentioned KEP). After health check endpoint is implemented, in dapr-cli, instead of grepping daprd output text, can use GET http://localhost:daprPort/health to verify daprd up healthy, then start app?

Reply:
The health check API seems to be a better approach IMO. This seems to be similar to discussion in https://github.com/dapr/dapr/issues/1288

Reply:
> The health check API seems to be a better approach IMO. This seems to be similar to discussion in [dapr/dapr#1288](https://github.com/dapr/dapr/issues/1288)

yes, #1288 solves such problems.